core/templating/datasource_query.go

```diff
@@ -0,0 +1,331 @@
+	case "SELECT":
+		selectRegex = regexp.MustCompile(`(?i)^SELECT\s+(.+)\s+FROM\s+([\w-]+)\s*(?:WHERE\s+(.+))?$`)
+		matches = selectRegex.FindStringSubmatch(query)
+		for i, _ := range matches {
+			log.Debug(fmt.Sprint(":", matches[i]))
+		}
+
+		if len(matches) < 3 {
+
+			return SQLStatement{}, errors.New("invalid query format")
+
+		}
+		columnsPart = matches[1]
+		dataSourceName = matches[2]
+		if !dataSourceExists(datasource, dataSourceName) {
+			return SQLStatement{}, errors.New("data source does not exist")
+		}
+		//if len(matches) == 4 && len(matches[3]) > 0 {
+		if len(matches) == 4 {
+			wherePart = matches[3]
+		}
+		headers := datasource[dataSourceName].Data[0]
+		columns := parseColumns(columnsPart, headers)
+
+		conditions, err := parseConditions(wherePart)
+		if err != nil {
+			return SQLStatement{}, err
+		}
+
+		return SQLStatement{
+			Type:           queryType,
+			Columns:        columns,
+			Conditions:     conditions,
+			DataSourceName: dataSourceName,
+		}, nil
+	case "UPDATE":
+		selectRegex = regexp.MustCompile(`(?i)^UPDATE\s+([\w-]+)\s+SET\s+(.+?)(?:\s+WHERE\s+(.+))?$`)
+		matches = selectRegex.FindStringSubmatch(query)
+		if len(matches) < 3 {
+			return SQLStatement{}, errors.New("invalid UPDATE query format")
+		}
+		dataSourceName = matches[1]
+		if !dataSourceExists(datasource, dataSourceName) {
+			return SQLStatement{}, errors.New("data source does not exist")
+		}
+		setPart := matches[2]
+		if len(matches) == 4 {
+			wherePart = matches[3]
+		}
+
+		setClauses := parseSetClauses(setPart)
+		conditions, err := parseConditions(wherePart)
+		if err != nil {
+			return SQLStatement{}, err
+		}
+		return SQLStatement{
+			Type:           "UPDATE",
+			Conditions:     conditions,
+			SetClauses:     setClauses,
+			DataSourceName: dataSourceName,
+		}, nil
+
+	case "DELETE":
+		selectRegex = regexp.MustCompile(`(?i)^DELETE\s+FROM\s+([\w-]+)\s*(?:WHERE\s+(.+))?$`)
+		matches = selectRegex.FindStringSubmatch(query)
+		if len(matches) < 2 {
+			return SQLStatement{}, errors.New("invalid DELETE query format")
+		}
+		dataSourceName = matches[1]
+		if !dataSourceExists(datasource, dataSourceName) {
+			return SQLStatement{}, errors.New("data source does not exist")
+		}
+		if len(matches) == 3 {
+			wherePart = matches[2]
+		}
+		conditions, err := parseConditions(wherePart)
+		if err != nil {
+			return SQLStatement{}, err
+		}
+		return SQLStatement{
+			Type:           "DELETE",
+			Conditions:     conditions,
+			DataSourceName: dataSourceName,
+		}, nil
+	}
+	return SQLStatement{}, errors.New("invalid query format")
+}
+
+// parseColumns determines the columns to select based on the query part and headers
+func parseColumns(columnsPart string, headers []string) []string {
+	columnsPart = strings.TrimSpace(columnsPart)
+	if columnsPart == "*" {
+		return headers
+	}
+	columns := strings.Split(columnsPart, ",")
+	for i, column := range columns {
+		columns[i] = strings.TrimSpace(column)
+	}
+	return columns
```

Comment:
it's not validating if the column a valid header or not

Reply:
Done